src/parser/mod.rs

```diff
@@ -6193,24 +6186,160 @@ impl<'a> Parser<'a> {
             && dialect_of!(self is MySqlDialect | SQLiteDialect | DuckDbDialect | GenericDialect)
         {
             self.parse_optional_column_option_as()
+        } else if self.parse_keyword(Keyword::AUTOINCREMENT)
+            && dialect_of!(self is SnowflakeDialect | SQLiteDialect |  GenericDialect)
+        {
+            if dialect_of!(self is SnowflakeDialect) {
```

Comment:
One general comment, we recently started to discourage the use of `dialect_of` macro in the code base. See [here](https://github.com/apache/datafusion-sqlparser-rs/blob/1e0460a/src/dialect/mod.rs#L63-L68). Can we convert the usages here to use the dialect method as described in the linked doc where possible?

To keep the scope limited to the PR's concerns I'd suggest we only look to convert the new introductions, like the current line. In the places where we're adding snowflake as an extra variant to the dialect_of check like here
```
if dialect_of!(self is SnowflakeDialect | SQLiteDialect | GenericDialect)
```
we can keep as is, since trying to convert such usages would be out of scope for this PR I think

Reply:
Oh actually, another suggestion I'm thinking makes more sense. the current code has a lot of snowflake specific guards now which isn't ideal, I'm thinking now it's worth doing similar to what we have for when parsing statements like create table, where we shell out to the dialect to handle any custom syntax. That would limit the changes to this file, and have the new logic live within the snowflake dialect.
See the [dialect.parse_statement()](https://github.com/apache/datafusion-sqlparser-rs/blob/2f076ed84195e2d9798f549ef18d95f7ae8fb49c/src/parser/mod.rs#L465-L468) for what it looks like.
Essentially to add a similar `dialect.parse_column_option()` and call the function here beforehand is what I'm thinking?

Reply:
I moved all snowflake specific code of column options parsing into snowflake dialect implementation.